### PR TITLE
Add tests for background jobs modes

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
This PR adds more tests for cliBackground background jobs modes using occ commands.

Related issue: #33261

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 1]
```diff
@@ -411,6 +411,94 @@ public function administratorEnablesTheUserUsingTheOccCommand($username) {
...
+	 * @return void
+	 */
+	public function theAdministratorSetsLogLevelUsingTheOccCommand($level) {
+		$this->featureContext->invokingTheCommand(
+			"log:manage --level $level"
+		);
+	}
+	
+	/**
+	 * @When the administrator sets the timezone to :timezone using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: Maybe this is just the timezone for the log file/backend?

```
When the administrator sets the log timezone to :timezone using the occ command
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 2]
```diff
@@ -411,6 +411,94 @@ public function administratorEnablesTheUserUsingTheOccCommand($username) {
...
+	 * @return void
+	 */
+	public function theAdministratorSetsTimeZoneUsingTheOccCommand($timezone) {
+		$this->featureContext->invokingTheCommand(
+			"log:manage --timezone $timezone"
+		);
+	}
+	
+	/**
+	 * @When the administrator sets the backend to :backend using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: Maybe there are "backends" of other things, as well as the log.
```
When the administrator sets the log backend to :backend using the occ command
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 3]
```diff
@@ -411,6 +411,94 @@ public function administratorEnablesTheUserUsingTheOccCommand($username) {
...
+	 * @return void
+	 */
+	public function theAdministratorSetsBackendUsingTheOccCommand($backend) {
+		$this->featureContext->invokingTheCommand(
+			"log:manage --backend $backend"
+		);
+	}
+	
+	/**
+	 * @When the administrator enables the ownCloud backend using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]: I think this is enabling the log?
```
When the administrator enables the ownCloud log using the occ command
```

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -411,6 +411,94 @@ public function administratorEnablesTheUserUsingTheOccCommand($username) {
...
+	 * @return void
+	 */
+	public function theAdministratorSetsLogFilePathUsingTheOccCommand($path) {
+		$this->featureContext->invokingTheCommand(
+			"log:owncloud --file $path"
+		);
+	}
+	
+	/**
+	 * @When the administrator sets the file size to :size for using the occ command
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```
@When the administrator sets the log file size to :size for using the occ command
```

File: tests/acceptance/features/cliBackground/backgroundJobs.feature [hunk 5]
```diff
@@ -0,0 +1,32 @@
+@cli @skipOnLDAP
+Feature: change background jobs mode
+  As an admin
+  I want to be able to change background jobs modes
+  So that I use different background jobs
+
+  Scenario: change background jobs mobe to cron
```
Comment (phil-davis) [inline tests/acceptance/features/cliBackground/backgroundJobs.feature]: s/mobe/mode/g

**Review by phil-davis — COMMENTED** [review]
I see now that a lot of those log commands were already there, and somehow moved in the diff.
Anyway, maybe we should improve them while we notice them.

**Comment by phil-davis** [comment]
We also need to think how many new test suites we want for all these small ``occ`` commands. Let's have a short white-board session in the morning to sort out grouping them into just a few suites.

**Comment by paurakhsharma** [comment]
@phil-davis, Can you please review it again.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@paurakhsharma now there are conflicts!

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@paurakhsharma next conflicts are ready to resolve

**Review by phil-davis — APPROVED** [review]

**Comment by paurakhsharma** [comment]
Backport on: #33281

**Comment by phil-davis** [comment]
New backport in PR #33285

**Pull request merged**